Use PROJECT_BINARY_DIR which is the true build dir for cmake.
===================================================================
Index: HIP-rocm-5.4.3/tests/hit/HIT.cmake
===================================================================
--- HIP-rocm-5.4.3.orig/tests/hit/HIT.cmake
+++ HIP-rocm-5.4.3/tests/hit/HIT.cmake
@@ -413,7 +413,7 @@ macro(HIT_ADD_FILES _config _dir _label
                         set_target_properties(${target} PROPERTIES OUTPUT_NAME ${_target_r} RUNTIME_OUTPUT_DIRECTORY "." LINK_DEPENDS "${HIP_LIB_FILES}" PREFIX "" SUFFIX "")
                 else()
                     # message(STATUS "add_custom_target*: target= ${target}  _buildcmd= ${_buildcmd}")
-                    add_custom_target(${target} COMMAND sh -c "${_buildcmd} -L${CMAKE_CURRENT_SOURCE_DIR}/build/lib -isystem ${CMAKE_CURRENT_SOURCE_DIR}/include")
+                    add_custom_target(${target} COMMAND sh -c "${_buildcmd} -L${PROJECT_BINARY_DIR}/lib -isystem ${CMAKE_CURRENT_SOURCE_DIR}/include")
                 endif()
                 add_dependencies(${_parent} ${target})
                 foreach(_dependency ${_depends})
